jolt-core/src/jolt/vm/mod.rs

```diff
@@ -344,135 +353,177 @@ where
 
     #[tracing::instrument(skip_all, name = "Jolt::prove_r1cs")]
     fn prove_r1cs(
+        preprocessing: JoltPreprocessing<F, G>,
         instructions: Vec<Self::InstructionSet>,
         bytecode_rows: Vec<ELFRow>,
         trace: Vec<ELFRow>,
         bytecode: Vec<ELFInstruction>,
         memory_trace: Vec<MemoryOp>,
         circuit_flags: Vec<F>,
+        jolt_polynomials: &JoltPolynomials<F, G>,
+        jolt_commitments: &JoltCommitments<G>,
         transcript: &mut Transcript,
     ) -> R1CSProof {
         let N_FLAGS = 17;
         let TRACE_LEN = trace.len();
+        let PADDED_TRACE_LEN = TRACE_LEN.next_power_of_two();
 
         let log_M = log2(M) as usize;
 
-        let [mut prog_a_rw, mut prog_v_rw, _] =
-            BytecodePolynomials::<F, G>::r1cs_polys_from_bytecode(bytecode_rows, trace);
+        let hyrax_generators: HyraxGenerators<NUM_R1CS_POLYS, G> =
```

Comment:
Setting NUM_R1CS_POLYS to 1 here is a hack, let's fix more directly before merging.